File: CM.Backend/CM.Backend.Application/Models/Dashboard/DashboardInfoDto.cs
namespace CM.Backend.Application.Models.Dashboard
{
    public class DashboardInfoDto
    {
        public int CattleTotal { get; set; }
        public List<GroupInfo> GroupInfos { get; set; } = new List<GroupInfo>();
        public List<PenInfo> PenInfos { get; set; } = new List<PenInfo>();
        public DailyMilking DailyMilking { get; set; }
        public DailyTasks DailyTasks { get; set; }
    }

    public class DailyTasks
    {
        public int TotalToDo { get; set; }
        public int TotalOngoing { get; set; }
        public int TotalDone {  get; set; }
        public Dictionary<string, string> TaskStatusList {  get; set; } = new Dictionary<string, string>();
    }

    public class DailyMilking
    {
        public int TotalToDo { get; set; }
        public int TotalDone {  get; set; }
        public double Volume { get; set; }
    }

    public class GroupInfo
    {
        public string Name {  get; set; }
        public int Total { get; set; }
    }

    public class PenInfo
    {
        public string Name {  get; set; }
        public int Size { get; set; }
        public int Total { get; set; }
    }
}

File: CM.Portal/src/app/api/models/dashboard-info-dto.ts
/* tslint:disable */
/* eslint-disable */
import { DailyMilking } from '../models/daily-milking';
import { DailyTasks } from '../models/daily-tasks';
import { GroupInfo } from '../models/group-info';
import { PenInfo } from '../models/pen-info';
export interface DashboardInfoDto {
  cattleTotal?: number;
  dailyMilking?: DailyMilking;
  dailyTasks?: DailyTasks;
  groupInfos?: Array<GroupInfo> | null;
  penInfos?: Array<PenInfo> | null;
}
